### PR TITLE
Non-interactive exec: trusted callers skip the prompt; --yes/VAULTKEEPER_YES for CI

### PR DESCRIPTION
**Pull request by mike-north**
Refs #58

## What

Makes `vaultkeeper exec` usable non-interactively (CI/Docker), building on the TOFU trust gate from #79.

- **Trusted callers skip the prompt** on a TTY or not (already held post-#79; now locked with a subprocess test).
- **`--yes` flag and `VAULTKEEPER_YES=1`** approve an untrusted caller for a single invocation without prompting, recording the approval via `setup()`'s TOFU first-encounter — exactly as an interactive `y` would. An explicit, documented opt-in; never the default.
- **Non-TTY, untrusted, no `--yes`** still fails, but the error now names the exact remediation (`vaultkeeper approve --script <caller>` or `--yes`).
- **A changed binary (identity mismatch) is never auto-approved by `--yes`** — it must be re-approved with `vaultkeeper approve`.
- `exec --help` documents the TTY requirement and both escape hatches; README gains a "Running in CI" note.

## Acceptance criteria → tests

| Criterion | Test |
|---|---|
| 1. Trusted caller, no prompt (TTY/non-TTY) | e2e `does not prompt for an approved caller and injects the secret` |
| 2. `--yes` approves + records trust | e2e `approves an untrusted caller with --yes and records trust for later runs`; unit `approves an untrusted caller non-interactively with --yes and records trust via setup` |
| 2. `VAULTKEEPER_YES=1` equivalent | e2e `approves an untrusted caller via VAULTKEEPER_YES=1 and records it in the manifest`; unit `approves an untrusted caller non-interactively via VAULTKEEPER_YES=1` |
| 3. Non-TTY untrusted fails with remediation | e2e `fails an untrusted caller on non-TTY stdin with actionable remediation`; unit `fails with remediation guidance for an untrusted caller on non-TTY stdin` |
| 4. `exec --help` documents TTY + escape hatches | e2e `documents the TTY requirement and escape hatches in exec --help`; unit help tests |
| 5. Subprocess regression tests | all e2e tests run the real CLI with piped stdin under isolated HOME + file backend |
| 6. README "Running in CI" note | `README.md` |

Also added `shouldAutoApprove` unit tests covering the strict `=1` env semantics (negative/boundary cases).

## `vaultkeeper exec --help`: Options and approval guidance

Documents the new `--yes` flag, the TTY requirement, and `VAULTKEEPER_YES`.

```diff
   --reason <text>    Human-readable reason for access (optional)
+  --yes              Approve an untrusted caller for this invocation without
+                     an interactive prompt, recording it in the trust manifest
+                     (for CI/non-interactive use; never the default)
   --cache            Cache the JWE token for subsequent invocations
   --no-redact        Do not redact the secret from output
   --skip-doctor      Skip doctor preflight checks
   -h, --help         Show this help message

+Approval and TTY requirement:
+  A caller that is not yet trusted requires approval. On an interactive
+  terminal you are prompted [y/N]. With non-TTY stdin (CI, pipes) there is
+  no prompt, so you must either pre-approve the caller with
+  `vaultkeeper approve --script <caller>` or pass --yes (or VAULTKEEPER_YES=1)
+  to approve this invocation. Once trusted, exec never prompts again.
+
 Environment variables:
+  VAULTKEEPER_YES=1           Approve an untrusted caller non-interactively
+                              (same as --yes)
   VAULTKEEPER_SKIP_DOCTOR=1   Skip doctor preflight checks
```

## `exec` untrusted caller on non-TTY stdin: error message

Before this change the message gave no path forward; now it does.

```diff
-Error: Secret access requires interactive approval. Run this command in a terminal.
+Error: Secret access for <caller> requires approval, but stdin is not a TTY, so no interactive prompt can be shown.
+To approve non-interactively, either:
+  - pre-approve the caller once:  vaultkeeper approve --script <caller>
+  - or approve just this run:      re-run with --yes (or set VAULTKEEPER_YES=1)
```

## Quality gate

`pnpm check` (typecheck + lint + api-report) green; vaultkeeper 603, @vaultkeeper/cli 199, @vaultkeeper/cli-tests 55 passed. Changeset: minor, `@vaultkeeper/cli`.

## Non-goals

Rust CLI parity (tracked separately) and changes to the interactive prompt UX.

### CODE REVIEW

File: packages/cli/src/commands/exec.ts [hunk 1]
```diff
@@ -94,11 +128,22 @@ function printExecHelp(): void {
...
       '  --cache            Cache the JWE token for subsequent invocations\n' +
       '  --no-redact        Do not redact the secret from output\n' +
       '  --skip-doctor      Skip doctor preflight checks\n' +
       '  -h, --help         Show this help message\n\n' +
+      'Approval and TTY requirement:\n' +
+      '  A caller that is not yet trusted requires approval. On an interactive\n' +
+      '  terminal you are prompted [y/N]. With non-TTY stdin (CI, pipes) there is\n' +
+      '  no prompt, so you must either pre-approve the caller with\n' +
+      '  `vaultkeeper approve --script <caller>` or pass --yes (or VAULTKEEPER_YES=1)\n' +
+      '  to approve this invocation. Once trusted, exec never prompts again.\n\n' +
```
[the remark this answers is not in the document]
Comment (mike-north) [reply packages/cli/src/commands/exec.ts:161]: Fixed. Removed the backticks around the suggested command in exec --help; it now reads: pre-approve the caller with "vaultkeeper approve --script <caller>". Swept the rest of the help and remediation output — no other backtick-wrapped commands remain (the non-TTY remediation and identity-mismatch messages use plain text).

File: packages/cli/src/commands/exec.ts [hunk 2]
```diff
@@ -78,6 +91,27 @@ async function enforceTrustGate(
     )
   }
 
+  // Never-approved caller. --yes / VAULTKEEPER_YES approves this invocation
+  // without prompting; setup() then records the hash via TOFU just as an
+  // interactive "y" would.
+  if (autoApprove) {
+    process.stderr.write('Trust: approved via --yes (recording caller in trust manifest)\n')
+    return true
+  }
```
[the remark this answers is not in the document]
Comment (mike-north) [reply packages/cli/src/commands/exec.ts:115]: Good catch — real bug. The trust gate now returns a three-way outcome (trusted | approved | denied) instead of a boolean, and exec only reads a cached token when the caller was ALREADY trusted. A just-approved caller (--yes or interactive 'y') always goes through setup(), which records the hash via TOFU exactly as before — the cache (its dir is independent of the trust manifest) can no longer short-circuit that recording, so the fresh-HOME + stale-cache case now records trust as criterion 2 requires. A modified caller still throws IdentityMismatchError at the gate before any cache access. Added a regression test 'records trust via setup for a --yes caller even when a cached token exists' asserting the cache is not even consulted, setup() runs, and the fresh token (not the cached one) is authorized.

File: packages/cli/src/commands/exec.ts [hunk 3]
```diff
@@ -78,6 +91,27 @@ async function enforceTrustGate(
...
+  // guidance instead of the raw "requires interactive approval" error.
+  // isTTY is `true` only on a real terminal; `undefined`/`false` means non-TTY.
+  if (!process.stdin.isTTY) {
+    throw new Error(
+      `Secret access for ${callerPath} requires approval, but stdin is not a TTY, ` +
+        `so no interactive prompt can be shown.\n` +
+        `To approve non-interactively, either:\n` +
+        `  - pre-approve the caller once:  vaultkeeper approve --script ${callerPath}\n` +
+        `  - or approve just this run:      re-run with --yes (or set VAULTKEEPER_YES=1)`,
+    )
```
[the remark this answers is not in the document]
Comment (mike-north) [reply packages/cli/src/commands/exec.ts:128]: Fixed. Added a shellQuote helper (POSIX single-quoting with '\'' escaping for embedded quotes) and applied it to callerPath in both remediation strings — the identity-mismatch guidance and the non-TTY remediation. A path with spaces or metacharacters ($(), ;, &, backticks, embedded quotes) is now emitted single-quoted and safe to copy/paste. Covered by a shell-quote unit test (spaces, metacharacters, embedded/multiple single quotes, empty string); the exec unit + e2e assertions were updated to expect the quoted form.
